docs/static/config-details.asciidoc

```diff
@@ -118,6 +118,34 @@ To summarize, we have 3 categories of memory usage, where 2 can be limited by th
 
 Keep these memory requirements in mind as you calculate your ideal memory allocation.
 
+[[reducing-off-heap-usage]]
+===== Troubleshooting out of memory errors in off-heap space: `pipeline.buffer.type`
+
+Plugins such as {agent}, {beats}, TCP, and HTTP inputs, use direct memory to get
+better performance when interacting with the network stack. 
+Under certain circumstances of heavy load, the direct memory space can be exhausted and lead to Out of Memory (OOM) errors in off-heap space.
+
+{ls} provides a `pipeline.buffer.type` setting in <<logstash-settings-file>> that can help you investigate and troubleshoot Out of Memory errors. 
+This setting lets you control where to allocate memory buffers for plugins that use them. 
+It is set to `direct` by default, but you can change it to `heap` to use Java heap space instead.
+When set to `heap`, all IO buffer allocations used by plugins are configured to **prefer** the
+Java Heap instead of direct memory. 
```

Comment:
```suggestion
An off-heap OOM is difficult to debug, so {ls} provides a `pipeline.buffer.type` setting in <<logstash-settings-file>> that lets you control where to allocate memory buffers for plugins that use them. 
Currently it is set to `direct` by default, but you can change it to `heap` to use Java heap space instead, which will be become the default in the future.
When set to `heap`, buffer allocations used by plugins are configured to **prefer** the
Java Heap instead of direct memory, as direct memory allocations may still be necessary depending on the plugin.
```